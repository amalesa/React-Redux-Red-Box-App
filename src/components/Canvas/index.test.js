import React from 'react';
import { shallow } from 'enzyme';
import { Canvas } from './index';
import yAxisValue from '../../reducers/yAxis';
import xAxisValue from '../../reducers/xAxis';

const props = {
  addYaxis: jest.fn(),
  minusYaxis: jest.fn(),
  addXaxis: jest.fn(),
  minusXaxis: jest.fn(),
  yAxisValue: 0,
  xAxisValue: 0,
};

const canvas = shallow(<Canvas {...props} />);

describe('<Canvas />', () => {
  it('renders', () => {
    expect(canvas).toMatchSnapshot();
  });

  it('displays the axis from props', () => {
    expect(canvas.find('.red-box-container').text()).toEqual('y: 0, x: 0');
  });
});

describe('when the user clicks the up event', () => {
  beforeEach(() => {
    canvas
      .find('.button.up')
      .simulate('click');
  });

  it('minusYaxis should be called', () => {
    expect(props.minusYaxis).toHaveBeenCalled();
  });
});

describe('when the user clicks the up event', () => {
  beforeEach(() => {
    canvas
      .find('.button.down')
      .simulate('click');
  });

  it('minusYaxis should be called', () => {
    expect(props.minusYaxis).toHaveBeenCalled();
  });
});

describe('click event', () => {
  it('on click down', () => {
    canvas
      .find('.button.down')
      .simulate('click');
    expect(props.minusYaxis).toHaveBeenCalled();
  });

  it('on click left', () => {
    canvas
      .find('.button.left')
      .simulate('click');
    expect(props.minusXaxis).toHaveBeenCalled();
  });

  it('on click right', () => {
    canvas
      .find('.button.right')
      .simulate('click');
    expect(props.addXaxis).toHaveBeenCalled();
  });
});

describe('click upevent', () => {
  it('on click down', () => {
    canvas
      .find('.button.down')
      .simulate('click');
    expect(props.minusYaxis).toHaveBeenCalled();
  });

  it('on click left', () => {
    canvas
      .find('.button.left')
      .simulate('click');
    expect(props.minusXaxis).toHaveBeenCalled();
  });

  it('on click right', () => {
    canvas
      .find('.button.right')
      .simulate('click');
    expect(props.addXaxis).toHaveBeenCalled();
  });
});
