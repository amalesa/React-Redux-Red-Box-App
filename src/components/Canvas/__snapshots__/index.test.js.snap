// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Canvas /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Canvas
    addXaxis={[Function]}
    addYaxis={[Function]}
    minusXaxis={[Function]}
    minusYaxis={[Function]}
    xAxisValue={0}
    yAxisValue={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="red-box-container"
          style={
                    Object {
                              "left": 0,
                              "top": 0,
                            }
          }
>
          y: 
          0
          , x: 
          0
          <div
                    className="red-box"
          />
</div>,
        <div
          className="canvas-board"
/>,
        <div
          className="controll-pannel"
>
          <div
                    className="button up"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
          >
                    UP
          </div>
          <div
                    className="button down"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
          >
                    DOWN
          </div>
          <div
                    className="button left"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
          >
                    LEFT
          </div>
          <div
                    className="button right"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
          >
                    RIGHT
          </div>
</div>,
      ],
      "className": "canvas",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "y: ",
            0,
            ", x: ",
            0,
            <div
              className="red-box"
/>,
          ],
          "className": "red-box-container",
          "style": Object {
            "left": 0,
            "top": 0,
          },
        },
        "ref": null,
        "rendered": Array [
          "y: ",
          0,
          ", x: ",
          0,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "red-box",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "canvas-board",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="button up"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
>
              UP
</div>,
            <div
              className="button down"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
>
              DOWN
</div>,
            <div
              className="button left"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
>
              LEFT
</div>,
            <div
              className="button right"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
>
              RIGHT
</div>,
          ],
          "className": "controll-pannel",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "UP",
              "className": "button up",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
            },
            "ref": null,
            "rendered": "UP",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "DOWN",
              "className": "button down",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
            },
            "ref": null,
            "rendered": "DOWN",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "LEFT",
              "className": "button left",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
            },
            "ref": null,
            "rendered": "LEFT",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "RIGHT",
              "className": "button right",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
            },
            "ref": null,
            "rendered": "RIGHT",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="red-box-container"
            style={
                        Object {
                                    "left": 0,
                                    "top": 0,
                                  }
            }
>
            y: 
            0
            , x: 
            0
            <div
                        className="red-box"
            />
</div>,
          <div
            className="canvas-board"
/>,
          <div
            className="controll-pannel"
>
            <div
                        className="button up"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
            >
                        UP
            </div>
            <div
                        className="button down"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
            >
                        DOWN
            </div>
            <div
                        className="button left"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
            >
                        LEFT
            </div>
            <div
                        className="button right"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
            >
                        RIGHT
            </div>
</div>,
        ],
        "className": "canvas",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "y: ",
              0,
              ", x: ",
              0,
              <div
                className="red-box"
/>,
            ],
            "className": "red-box-container",
            "style": Object {
              "left": 0,
              "top": 0,
            },
          },
          "ref": null,
          "rendered": Array [
            "y: ",
            0,
            ", x: ",
            0,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "red-box",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "canvas-board",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="button up"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
>
                UP
</div>,
              <div
                className="button down"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
>
                DOWN
</div>,
              <div
                className="button left"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
>
                LEFT
</div>,
              <div
                className="button right"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
>
                RIGHT
</div>,
            ],
            "className": "controll-pannel",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "UP",
                "className": "button up",
                "onClick": [Function],
                "onKeyDown": [Function],
                "role": "button",
              },
              "ref": null,
              "rendered": "UP",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "DOWN",
                "className": "button down",
                "onClick": [Function],
                "onKeyDown": [Function],
                "role": "button",
              },
              "ref": null,
              "rendered": "DOWN",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "LEFT",
                "className": "button left",
                "onClick": [Function],
                "onKeyDown": [Function],
                "role": "button",
              },
              "ref": null,
              "rendered": "LEFT",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "RIGHT",
                "className": "button right",
                "onClick": [Function],
                "onKeyDown": [Function],
                "role": "button",
              },
              "ref": null,
              "rendered": "RIGHT",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
